nistats/glm.py

```diff
@@ -58,183 +47,394 @@ def data_scaling(Y):
+             memory=Memory(None), memory_level=1, standardize=False,
+             percent_signal_change=True, verbose=1, n_jobs=1,
+             noise_model='ar1'):
+        self.mask = mask
+        self.memory = memory
+        self.memory_level = memory_level
+        self.verbose = verbose
+        self.standardize = standardize
+        self.n_jobs = n_jobs
+        self.low_pass = low_pass
+        self.high_pass = high_pass
+        self.t_r = t_r
+        self.target_affine = target_affine
+        self.target_shape = target_shape
+        self.smoothing_fwhm = smoothing_fwhm
+        self.noise_model = noise_model
+        self.percent_signal_change = percent_signal_change
+        if self.percent_signal_change:
+            self.standardize = False
+
+    def fit(self, imgs, design_matrices):
+        """ Note: design_matrices is the design matrix !
+        1. does a masker job: fMRI_data -> Y
+        2. fit an ols regression to (Y, X)
+        3. fit an AR(1) regression of require
+        This results in an internal (labels_, regression_results_) parameters
 
         Parameters
         ----------
-        column_index : int or array-like of int or None, optional
-            The indexed of the columns to be returned.  if None (default
-            behaviour), the whole vector is returned
+        imgs: Niimg-like object or list of Niimg-like objects,
+            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
+            Data on which the GLM will be fitted. If this is a list,
+            the affine is considered the same for all.
 
-        Returns
-        -------
-        beta: array of shape (n_voxels, n_columns)
-            the required coefficients
-        """
-        # make colum_index a list if it an int
-        if column_index is None:
-            column_index = np.arange(self.X.shape[1])
-        if not hasattr(column_index, '__iter__'):
-            column_index = [int(column_index)]
-        n_beta = len(column_index)
-
-        # build the beta array
-        beta = np.zeros((n_beta, self.labels_.size), dtype=np.float)
-        for l in self.results_.keys():
-            beta[:, self.labels_ == l] = self.results_[l].theta[column_index]
-        return beta
-
-    def get_mse(self):
-        """Acessor for the mean squared error of the model
+        design_matrices: array or list of arrays
+            Models of the temporal events considered in the analysis.
+            the arrays have shape (n_time_points, n_regressors)
 
-        Returns
-        -------
-        mse: array of shape (n_voxels)
-            the sum of square error per voxel
         """
-        # build the beta array
-        mse = np.zeros(self.labels_.size, dtype=np.float)
-        for l in self.results_.keys():
-            mse[self.labels_ == l] = self.results_[l].MSE
-        return mse
+        # First, learn the mask
+        if not isinstance(self.mask, NiftiMasker):
```

Comment:
@arthurmensch it is not what your check_embedded_masker or something was doing ? I guess it is only in nilearn master right?